Make frame format introductions consistent

martinthomson opened the pull request:
All references to figures that describe frame formats are changed to use
the form: "X frames are formatted as shown in {{x}}."

This retains some of the extra details so that the change is limited to
this change, but I made the empty frame description consistent
throughout (with a little extra explanation for PADDING, which is
first).

Closes #3804.

MikeBishop commented:
This is a good improvement, but I disagree that it addresses #3804.  That is specifically about the section headings, in which some are "X Frame" and others are "X Frames":

> 19. Frame Types and Formats
> 19.1.  PADDING Frame
> 19.2.  PING Frame
> **19.3.  ACK Frames**
> 19.4.  RESET_STREAM Frame
> 19.5.  STOP_SENDING Frame
> 19.6.  CRYPTO Frame
> 19.7.  NEW_TOKEN Frame
> **19.8.  STREAM Frames**
> 19.9.  MAX_DATA Frame
> 19.10. MAX_STREAM_DATA Frame
> **19.11. MAX_STREAMS Frames**
> 19.12. DATA_BLOCKED Frame
> 19.13. STREAM_DATA_BLOCKED Frame
> **19.14. STREAMS_BLOCKED Frames**
> 19.15. NEW_CONNECTION_ID Frame
> 19.16. RETIRE_CONNECTION_ID Frame
> 19.17. PATH_CHALLENGE Frame
> 19.18. PATH_RESPONSE Frame
> **19.19. CONNECTION_CLOSE Frames**
> 19.20. HANDSHAKE_DONE Frame
> 19.21. Extension Frames

martinthomson commented:
My bad.  I had forgotten that bit.  I made these plural and changed definitions from "the X frame" to "a[n] X frame".  For the multi-valued types, we were also inconsistent with use of "or" vs. "and", so I changed that to match.
